hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -933,7 +933,7 @@ private NonSelectQueryPlan buildInsertQueryPlan() {
 		final EntityPersister entityDescriptor = getSessionFactory().getDomainModel().findEntityDescriptor( entityNameToInsert );
 
 		final SqmMultiTableInsertStrategy multiTableStrategy = entityDescriptor.getSqmMultiTableInsertStrategy();
-		if ( multiTableStrategy == null || sqmInsert instanceof SqmInsertValuesStatement ) {
```

Comment:
Did you also fix the bug with `insert .... values`?

Reply:
I didn't encounter issues when running tests locally, so I wanted to check how it runs on CI. I'll look into that though if I see errors.

Reply:
Well yeah, of course, you probably didn't run the documentation tests.

Reply:
I did run the full suite with H2 and it seems H2 also works on CI, but there are errors for other DBs which I will look into tomorrow.

Reply:
And yes: if you check the build, `HQLTest.hql_insert_example()` is failing.

It's like you think I'm an incompetent idiot who just reports and fixes imaginary bugs.

I just don't understand.